src/vaadin-number-field.html

```diff
@@ -201,6 +226,28 @@
           }
           return super.checkValidity();
         }
+
+        _onInput(e) {
+          const input = this.focusElement;
+          if (e.data === '-' && (this.__previousValidInput === '-' || !this.__previousValidInput)) {
+            // Allow minus sign (-) as the first character
+            this.__previousValidInput = '-';
```

Comment:
None of the logic inside this method is tested!

Reply:
Removed this logic as I implemented the "option 2".